packages/stash/src/actions/extend.ts

```diff
@@ -0,0 +1,15 @@
+import { Stash } from "../common";
+
+export type ExtendArgs<stash extends Stash, actions> = {
+  stash: stash;
+  actions: actions;
+};
+
+export type ExtendResult<stash extends Stash, actions> = stash & actions;
```

Comment:
I guess the expectation/assumption here is there are no overlapping keys?

Reply:
you would?

https://www.typescriptlang.org/play/?#code/AQFwngDgpsCiAeIoDsAmAlKBnArgGxAB4sQBDLACwBphSBjEASwHtksA+YAXmBPIuAAyWgxZsA3AChJoSDCiIUqKKm5xFaTLgKEA3sADmUEAGFWAM0YGAXMAAUASm6cAbs0aqAvjX1HTFq1tHZ2BkHABbACMoACdgT3ZpWWhDYzNkSwM1BSQ0FQBtAHI-dMzCgF1JAHoq4Dq6gD0AfkkgA

looks like you get a union for each action

I wonder if we wanna use `merge` from arktype instead?

Reply:
> That means it is now technically possible to override existing functions.

this was already happening at runtime, the types just better reflect it now